src/snowflake/snowpark/mock/_stored_procedure.py

```diff
@@ -31,11 +42,13 @@
 
 
 def sproc_types_are_compatible(x, y):
-    if isinstance(x, type(y)):
-        return True
-    elif isinstance(x, _IntegralType) and isinstance(y, _IntegralType):
-        return True
-    elif isinstance(x, _FractionalType) and isinstance(y, _FractionalType):
+    if (
```

Comment:
When we work on coercion, this function should be replaced by like `DataType.can_coerce_to(DataType)`. @sfc-gh-aling